web/src/lib/components/asset-viewer/actions/rotate-action.svelte

```diff
@@ -0,0 +1,82 @@
+<script lang="ts">
+  import MenuOption from '$lib/components/shared-components/context-menu/menu-option.svelte';
+  import {
+    notificationController,
+    NotificationType,
+  } from '$lib/components/shared-components/notification/notification';
+  import { AssetAction } from '$lib/constants';
+  import { handleError } from '$lib/utils/handle-error';
+  import { AssetJobName, runAssetJobs, updateAsset, type AssetResponseDto } from '@immich/sdk';
+  import { mdiRotateLeft, mdiRotateRight } from '@mdi/js';
+  import { t } from 'svelte-i18n';
+  import type { OnAction } from './action';
+  import { photoViewer } from '$lib/stores/assets.store';
+
+  export let asset: AssetResponseDto;
+  export let onAction: OnAction;
+  export let to: 'left' | 'right';
+
+  const handleRotateAsset = async () => {
+    let newOrientation = '1';
+    try {
+      switch (asset.exifInfo?.orientation) {
```

Comment:
Can we use the enums here? And also can we just have a list and either find the next or previous element?

Reply:
We can. I was unsure whether using lists and stuff would be too "clever" and hinder readability so I wrote a dumb switch-case but I'll change it.

Reply:
I'd rather see a map or a list than a long case statement

Reply:
is this readable enough?
```ts
// every rotation, in order
// interleaved normal / mirrored
const orientations = ['1', '2', '6', '7', '3', '4', '8', '5'];
let index = orientations.indexOf(asset.exifInfo?.orientation ?? '1');
if (index === -1) {
  index = 0;
}
if (to === 'right') {
  index = (index + 2) % orientations.length;
} else {
  index = (index - 2 + orientations.length) % orientations.length;
}
```

Reply:
Seems fine to me. I probably would have preferred two lists and to use the enum instead of plain strings.

Reply:
I'm not sure how to use the enum, actually, since it's defined in the server package and this is the web app.

Reply:
done (but without the enum since it doesn't export the names)